marklogic-spark-connector/src/main/java/com/marklogic/spark/writer/DocBuilder.java

```diff
@@ -156,8 +160,12 @@ private DocumentWriteOperation buildMainDocument(DocumentInputs inputs) {
 
         AbstractWriteHandle content = inputs.getContent();
         byte[] classificationResponse = inputs.getClassificationResponse();
-        if (classificationResponse != null && inputs.getExtractedText() == null && Format.XML.equals(((BytesHandle) content).getFormat())) {
-            content = appendDocumentClassificationToXmlContent(((BytesHandle) content).get(), inputs.getInitialUri(), classificationResponse);
+        if (classificationResponse != null && inputs.getExtractedText() == null) {
+            if (Format.XML.equals(((BytesHandle) content).getFormat())) {
```

Comment:
For the format check - move `determineSourceDocumentFormat` in DefaultChunkAssembler into the Util class in the same subproject. Use that here to get the format. If it returns null, log a warning, just like DefaultChunkAssembler is doing.